### ISSUE TITLE
Queued offline actions not being triggered when network status changes

### ISSUE THREAD

**Issue by raultomescu** [redux-offline/redux-offline#165]
As the title says, my offline queued actions are not triggered when the network status changes to online. They are however triggered when any other action is triggered.

Offline config:

```js
const offlineConfig = {
  ...defaultConfig,
  persistOptions: {
    storage: localForage,
    whitelist: offlineWhiteList,
  },
  effect: async (actionEffect: Function) =>
    new Promise((resolve, reject) => {
      if (actionEffect) {
        actionEffect()
          .then(resolve)
          .catch(reject);
      } else {
        resolve();
      }
    }),
  discard: (_, retries: number) => retries === 100,
};
```

Offline action:

```js
export const someAction = () => ({
  type: actionTypes.TYPE,
  offline: async (client: any, {dispatch}: any) => ({
    effect: async () => {
       // client is a wrapper around axios
       await client.post();
    },
    commit: {type: actionTypes.TYPE_SUCCESS},
    rollback: {type: actionTypes.TYPE_FAIL},
  }),
});
```

Middleware:

```js
export function clientMiddleware(client) {
  return store => next => (action) => {
    const {dispatch, getState} = store;
    const {
      offline,
      ...rest,
     } = action;

    if (offline) {
      return offline(client, store).then(result => {
        return next({
          type: action.type,
          meta: {
            offline: result,
          },
        });
      });
    }
  };
}
```

**Comment by raultomescu** [comment]
I am just noticing this issue has been reported two times before: https://github.com/redux-offline/redux-offline/issues/132 and https://github.com/redux-offline/redux-offline/issues/146 .

In my case I cannot stringify the body of the effect, and being forced to do so enforces a big constraint. Why is it needed? Are the actions also persisted offline? If I had a stringified body and closed the app, would it retry after reopening it? 'cause that would be awesome! :)

But still, I think my use case is a valid one and I think others would benefit from a solution aswell. I am willing to work on it if someone can point me in the right direction and can answer questions from time to time.

Thanks!

**Comment by wacii** [comment]
@RaulTsc Stringifying is only needed with the default effect reconciler, as that uses the fetch api which doesn't stringify JSON for you like axios does.

I have looked at this issue before, and wrote a test for it, but was never able to reproduce the error. Granted your code deviates from the intended usage more than the other examples.

The only thing that jumps out is that you should set `config.persist` to false, as your use of functions in the offline actions means that this will never work between sessions. But that isn't going to cause this issue.

I'll give this another look.

**Comment by raultomescu** [comment]
Hey @wacii thanks for replying. Let me know if you need help.

So you are saying that if I use plain objects for the effect and implement my own effects reconciler (it needs auth and other stuff), the actions will be stored and retried between sessions?

**Comment by wacii** [comment]
@RaulTsc Yes, actions are retried between sessions. If they can be serialized.

**Comment by raultomescu** [comment]
Ok that's very interesting. I missed it while reading the docs - I thought it persists only the state.

**Comment by sorodrigo** [comment]
Solved in #229

**Comment by MartinJHammer** [comment]
Adding my fix example here as well (first added in https://github.com/Vizzuality/forest-watcher/pull/303)

The change fixing the problem happens in `composeEnhancers()`

My store creation before (actions not firing before other action is fired):

```typescript
const store: Store<IAppState> = createStore(
        offlineEnhancer.enhanceReducer(rootReducer),
        initialState,
        composeEnhancers(
            applyMiddleware(offlineEnhancer.middleware as any, epicMiddleware),
            offlineEnhancer.enhanceStore,
        )
    );
```
My store create after (offline queue now immediatly dispatches)

```typescript
   const store: Store<IAppState> = createStore(
        offlineEnhancer.enhanceReducer(rootReducer),
        initialState,
        composeEnhancers(
            offlineEnhancer.enhanceStore,
            applyMiddleware(offlineEnhancer.middleware as any, epicMiddleware),
        )
    );
```
